### PR TITLE
Enforce HTTPS on godoc.org

### LINKED ISSUE #304: Make godoc.org HTTPS-only
Is there a reason why http://godoc.org doesn't 301 to the https version? SSL is set up and works perfectly, seems like an easy fix.

### PR DESCRIPTION
**Pull request by garyburd**
Fixes #304.

### CODE REVIEW

**Comment by garyburd** [comment]
@jmhodges Please confirm that you are OK with your commit being contributed to Google.

File: gddo-server/main.go [hunk 1]
```diff
@@ -760,12 +761,30 @@ func handleAPIError(resp http.ResponseWriter, req *http.Request, status int, err
...
-func (m hostMux) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+func (m rootHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	host := req.Host
+	if h, _, err := net.SplitHostPort(host); err == nil {
+		host = h
+	}
+	if host == "godoc.org" {
+		// Because https is not used api.godoc.org, the includeSubDomains
+		// parameter is not used here.
+		resp.Header().Add("Strict-Transport-Security", "max-age=631138519; preload")
```
Comment (adg) [inline gddo-server/main.go]: Put this after the following if block (don't set the HSTS header on HTTP requests).

Comment (adg) [inline gddo-server/main.go]: Hey @agl does this header look right to you? I'm concerned about the length of the timeout. What's the best practice?

Comment (garyburd) [inline gddo-server/main.go]: I moved the code after the if block.

Comment (agl) [inline gddo-server/main.go]: > Hey @agl does this header look right to you? I'm concerned about the length of the timeout. What's the best practice?

The timeout should be at least 18 weeks (10886400). That value is 20
years, which is effectively infinity. It's certainly greater than 18
weeks ;)

**Pull request merged**